### ISSUE TITLE
Matching parameter not working for some events

### ISSUE THREAD

**Issue by razoarmani** [mIRC-Scripters/mSL-VSCode-Extension#20]
Hello, first time ever posting on github >_< not even sure if I'm doing this right... 

https://github.com/mIRC-Scripters/mSL-VSCode-Extension/blob/a292b33e74bff91381ef3c84a20ef196c8a60a14/syntaxes/msl.tmLanguage.json#L590

All the events in this match group support variables, aliases, evaluated variables, and aliases as well

> examples: 

```
set %var-chans #channel1,#channel2
alias -l alias_channels { return #channel1,#channel2 }

ON *:JOIN:%var-chans:
ON *:JOIN:$(%var-chans):
ON *:JOIN:$($alias_chans):
```

I did some testing from my side and replacing `(?:\\*|#[^:]*)` with `(?:\\*|(?:%|#|\\x24\\x28.+\\x29)[^:]*)` did the trick for me, maybe you can find a better way...

**Comment by Kedyn** [comment]
Welcome, thanks for posting.

When I created this, I mostly based myself in the mIRC Help File and the Wikichip mIRC Help Section. 

mSL syntax documentation is a bit of a struggle. SReject asked for help: https://forums.mirc.com/ubbthreads.php/topics/266539/help-documenting-syntax#Post266539

But only I responded with the little info I had from creating this and other projects similar to this one. So information like this is very helpful.
